_posts/2017-12-19-fuzzy-search-in-elixir.md

```diff
@@ -0,0 +1,284 @@
+---
+layout: post
+title:  "Implementing Fuzzy Search in Elixir Using Ecto and Postgres"
+date:   2017-12-19 12:00:00 -0400
+categories: api
+author: Demi Filippou
+---
+
+## Background
+Here at Forward Financing, our Elixir single sign on app has always stored only a small number of users. We developed only internal tools and therefore only kept track of internal users. However, now that we are expanding and developing external facing tools, the number of users is skyrocketing. This caused a problem for admins who were creating, editing, and checking for users. To check if a user exists, an admin would have to manually click through the growing number of pages of users. Yuck. That is why this sprint, our project managers decided it was time to implement search functionality for users. We decided to go down the route of fuzzy search, because why vanilla search when you can fuzzy search?
+
+### Where do we start?
+If we are going to write fuzzy search, we are going to need a way to calculate what strings are close enough to our query to show up in the results. After some quick investigating, we decided the best route to go down was to use the Levenshtein function from the [fuzzystrmatch Postgres module](https://www.postgresql.org/docs/current/static/fuzzystrmatch.html#idm46428636025488)
+
+### Levenshtein? What is that? Why are we using it?
+Levenshtein is often referred to as "edit distance," a name that better reflects its purpose. The Levenshtein distance between two strings is essentially the number of character changes to transform string A to string B. Valid character changes are:
```

Comment:
"changes are:"

You don't need a colon in this situation, since "are" acts as the verb on your list.